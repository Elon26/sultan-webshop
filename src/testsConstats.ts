export const testCatalog = [
    {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F1.jpg?alt=media&token=f1c105fe-7e4d-4e64-a891-9f344a4f4f99",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F1Small.jpg?alt=media&token=b6b8669a-5002-4228-952e-e353f4d166f1",
        name: "Жидкость для мытья посуды Garden Eco экологичный Цитрус",
        dimensionType: "мл",
        dimension: "500",
        barcode: "4600104028410",
        manufacturer: "Гармония",
        brand: "Garden",
        description:
            "Гель для мытья посуды легко удаляет остатки пищи, жир, пригоревшие загрязнения, налет от чая и кофе. Подходит для людей, склонных к аллергии. Может использоваться для мытья детской посуды.",
        price: 209,
        purposes: ["Для мытья посуды"]
    },
    {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F2.webp?alt=media&token=be27afb4-8f2f-4b0d-8e94-a01cf69855ed",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F2Small.jpg?alt=media&token=f0138e0e-896e-4cad-bf60-d80b19a77188",
        name: "Жидкость для мытья посуды Fairy Зеленое яблоко",
        dimensionType: "мл",
        dimension: "450",
        barcode: "4015400869320",
        manufacturer: "Procter & Gamble",
        brand: "Fairy",
        description:
            "Подходит для мытья стеклянной, обычной и детской посуды, фруктов и овощей, нержавеющей стали. Fairy - это безопасный продукт, который легко смывается водой и полностью соответствует ГОСТу.",
        price: 168,
        purposes: ["Для мытья посуды", "Для мытья фруктов"]
    },
    {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F3.jpg?alt=media&token=8b8c7476-6a1c-42e6-bd94-c4eab3fea7e3",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F3Small.jpg?alt=media&token=c8f720c4-241c-46e4-9057-c3d4ef2851f3",
        name: "Жидкость для мытья посуды Fairy Сочный лимон",
        dimensionType: "мл",
        dimension: "650",
        barcode: "4015400869412",
        manufacturer: "Procter & Gamble",
        brand: "Fairy",
        description:
            "Линейка Fairy Нежные руки разработана совместно с дерматологами для сохранения нежности кожи Ваших рук. Концентрированная формула Fairy быстро расщепляет жир и обеспечивает идеальную чистоту посуды, в то же время не провоцируя сухость кожи Ваших рук.",
        price: 223,
        purposes: ["Для мытья посуды", "Для мытья фруктов"]
    }
];

export const testTotalPriceAndQuantity = {
    totalPrice: 777777,
    totalQuantity: 42
};

export const defaultDataBase = {
    4600104028410: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F1.jpg?alt=media&token=f1c105fe-7e4d-4e64-a891-9f344a4f4f99",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F1Small.jpg?alt=media&token=b6b8669a-5002-4228-952e-e353f4d166f1",
        name: "Жидкость для мытья посуды Garden Eco экологичный Цитрус",
        dimensionType: "мл",
        dimension: "500",
        barcode: "4600104028410",
        manufacturer: "Гармония",
        brand: "Garden",
        description:
            "Гель для мытья посуды легко удаляет остатки пищи, жир, пригоревшие загрязнения, налет от чая и кофе. Подходит для людей, склонных к аллергии. Может использоваться для мытья детской посуды.",
        price: 209,
        purposes: ["Для мытья посуды"]
    },
    4015400869320: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F2.webp?alt=media&token=be27afb4-8f2f-4b0d-8e94-a01cf69855ed",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F2Small.jpg?alt=media&token=f0138e0e-896e-4cad-bf60-d80b19a77188",
        name: "Жидкость для мытья посуды Fairy Зеленое яблоко",
        dimensionType: "мл",
        dimension: "450",
        barcode: "4015400869320",
        manufacturer: "Procter & Gamble",
        brand: "Fairy",
        description:
            "Подходит для мытья стеклянной, обычной и детской посуды, фруктов и овощей, нержавеющей стали. Fairy - это безопасный продукт, который легко смывается водой и полностью соответствует ГОСТу.",
        price: 168,
        purposes: ["Для мытья посуды", "Для мытья фруктов"]
    },
    4015400869412: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F3.jpg?alt=media&token=8b8c7476-6a1c-42e6-bd94-c4eab3fea7e3",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F3Small.jpg?alt=media&token=c8f720c4-241c-46e4-9057-c3d4ef2851f3",
        name: "Жидкость для мытья посуды Fairy Сочный лимон",
        dimensionType: "мл",
        dimension: "650",
        barcode: "4015400869412",
        manufacturer: "Procter & Gamble",
        brand: "Fairy",
        description:
            "Линейка Fairy Нежные руки разработана совместно с дерматологами для сохранения нежности кожи Ваших рук. Концентрированная формула Fairy быстро расщепляет жир и обеспечивает идеальную чистоту посуды, в то же время не провоцируя сухость кожи Ваших рук.",
        price: 223,
        purposes: ["Для мытья посуды", "Для мытья фруктов"]
    },
    8001090017079: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F4.jpg?alt=media&token=0900ca15-0d64-4e20-b54a-df6bb531f336",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F4Small.jpg?alt=media&token=e00b964b-b976-4463-b22a-cbdd7b606b2b",
        name: "Жидкость для мытья посуды Fairy Нежные руки чайное дерево и мята",
        dimensionType: "мл",
        dimension: "900",
        barcode: "8001090017079",
        manufacturer: "Procter & Gamble",
        brand: "Fairy",
        description:
            "Концентрированная формула Fairy быстро расщепляет жир и обеспечивает идеальную чистоту посуды, в то же время не провоцируя сухость кожи Ваших рук. Активные компоненты формулы настолько эффективны, что запросто растворяют жир даже в холодной воде.",
        price: 272,
        purposes: ["Для мытья посуды", "Для мытья фруктов"]
    },
    4600697060613: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F5.jpg?alt=media&token=3fdad0d0-0e93-445a-89d3-044a15bc5718",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F5Small.jpg?alt=media&token=16f67ff8-b862-4b4f-a3cb-fa42543bffb9",
        name: "Гель для мытья посуды Сарма Антибактериальный Актив",
        dimensionType: "мл",
        dimension: "500",
        barcode: "4600697060613",
        manufacturer: "Невская Косметика",
        brand: "Сарма",
        description:
            "Гели прекрасно справляются с жиром даже в холодной воде. Обладают доказанным антибактериальным эффектом. Устраняют неприятные запахи. Экономно расходуются. Создают обильную густую пену, которая легко смывается с посуды, оставляя ее «скрипяще» чистой.",
        price: 85,
        purposes: ["Для мытья посуды"]
    },
    4084500644946: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F6.jpg?alt=media&token=4e2241ca-ebb6-4d83-baac-4549e8a9c50b",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F6Small.jpg?alt=media&token=30109a2c-54c8-4f35-9c36-dee9e563237e",
        name: "Моющее средство Mr.Proper для полов и стен Лавандовое спокойствие",
        dimensionType: "мл",
        dimension: "1000",
        barcode: "4084500644946",
        manufacturer: "Procter & Gamble",
        brand: "Mr.Proper",
        description:
            "Универсальное средство для уборки Mr. Proper с ароматом лаванды поможет справиться с загрязнениями дома. Моющая жидкость легко удаляет загрязнения с ламината, деревянного паркета, кафельной плитки и других твердых поверхностей.",
        price: 316,
        purposes: ["Для мытья полов", "Для мытья стен"]
    },
    4084500644991: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F7.jpg?alt=media&token=9a28e2dc-a713-4d16-bbe2-8edd55d93f57",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F7Small.jpg?alt=media&token=a6cbe5a8-f919-4eac-ae60-f4ce4a8ef7c5",
        name: "Моющее средство Mr.Proper для полов и стен Горный ручей и прохлада",
        dimensionType: "мл",
        dimension: "1000",
        barcode: "4084500644991",
        manufacturer: "Procter & Gamble",
        brand: "Mr.Proper",
        description:
            "Универсальное средство для уборки Mr. Proper с ароматом Горного ручья и прохлады поможет справиться с загрязнениями дома. Моющая жидкость легко удаляет загрязнения с ламината, деревянного паркета, кафельной плитки и других твердых поверхностей.",
        price: 320,
        purposes: ["Для мытья полов", "Для мытья стен"]
    },
    8711700855358: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F8.jpg?alt=media&token=bcb459e5-d0ff-4605-8219-4a476165d09b",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F8Small.jpg?alt=media&token=974f1d0b-05c7-4883-8533-5f684fddb73c",
        name: "Моющее средство Glorix Цветущая Яблоня и Ландыш для пола",
        dimensionType: "мл",
        dimension: "1000",
        barcode: "8711700855358",
        manufacturer: "Юнилевер Русь",
        brand: "Glorix",
        description:
            "Доверьте дезинфекцию пола Glorix! Наша бытовая химия не только прекрасно очищает любые поверхности, но и уничтожает микробы и вирусы. Мы заботимся как о здоровье своих покупателей, так и о природе, поэтому наша cleaners-продукция не содержит агрессивных веществ.",
        price: 402,
        purposes: ["Для мытья полов"]
    },
    4602984009848: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F9.jpg?alt=media&token=8cafb03d-1e1c-493c-ad5c-73057e148e52",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F9Small.jpg?alt=media&token=af85bf2a-b8a9-4fef-937a-4d37b3c1101e",
        name: "Средство для мытья пола Чистин Антибактериальный эффект Сочный лимон",
        dimensionType: "мл",
        dimension: "1000",
        barcode: "4602984009848",
        manufacturer: "Ступинский хим. завод",
        brand: "Чистин",
        description:
            "Уничтожает микробы, грибок, неприятные запахи. Предназначено для чистки и антимикробной обработки раковин, ванн, душевых кабин, унитазов, профилактики засоров, канализационных стоков, мытья керамической плитки, кухонных плит, любых твердых моющихся напольных покрытий, настенных панелей, моющихся обоев, бытовой техники.",
        price: 99,
        purposes: ["Для мытья полов"]
    },
    8001090628848: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F10.webp?alt=media&token=885570f3-820c-419d-ac40-3deb12e2b78e",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F10Small.jpg?alt=media&token=8514ddfa-9482-4e75-ac7b-b23dbd810d3f",
        name: "Моющее средство Mr.Proper Бережная уборка для полов и стен",
        dimensionType: "мл",
        dimension: "500",
        barcode: "8001090628848",
        manufacturer: "Procter & Gamble",
        brand: "Mr.Proper",
        description:
            "Сделайте уборку максимально простой и эффективной с моющей жидкостью Mr.Proper Бережная уборка. Универсальное средство подходит для бережного очищения пола, стен и корпусной мебели. Обеспечивает до 100% чистоты, не повреждает напольные покрытия и отличается ненавязчивым запахом.",
        price: 186,
        purposes: ["Для мытья полов", "Для мытья стен"]
    },
    4602984004867: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F11.jpg?alt=media&token=a680b66d-382c-45bb-be06-80f52b428d7c",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F11Small.jpg?alt=media&token=4fb50099-37b2-4151-af1f-00815e2e7d2d",
        name: "Чистящий спрей Sanita с нашатырным спиртом",
        dimensionType: "мл",
        dimension: "500",
        barcode: "4602984004867",
        manufacturer: "Ступинский хим. завод",
        brand: "Sanita",
        description:
            "Спрей для очистки стекол с нашатырным спиртом в составе рекомендован для мытья автомобильных стекол, витрин, окон, зеркал, кафеля, фаянса, наружных поверхностей холодильника. Для идеальной чистоты достаточно всего лишь нанести средство на поверхность и протереть. Не оставляет разводов и следов. Имеет приятный легкий цитрусовый аромат.",
        price: 195,
        purposes: ["Для мытья окон", "Для мытья зеркал"]
    },
    4601313011149: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F12.jpg?alt=media&token=1dc66fd4-192e-458b-8842-61d75dcd8c70",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F12Small.jpg?alt=media&token=981492ad-400c-4bf7-942e-383f7abee5d0",
        name: "Моющее средство Капля Vox для стекол и зеркал Цитрусовая свежесть",
        dimensionType: "мл",
        dimension: "500",
        barcode: "4601313011149",
        manufacturer: "Весна",
        brand: "Капля",
        description:
            "Средство для стекол и зеркал Капля VOX создано для быстрого и эффективного удаления загрязнений с окон и зеркальных поверхностей. Чистящее средство легко наносится при помощи мелкодисперсного пульверизатора, который равномерно распределяет средство на любой поверхности.",
        price: 181,
        purposes: ["Для мытья окон", "Для мытья зеркал"]
    },
    9000100888448: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F13.webp?alt=media&token=3daef6f9-1d99-4b54-9cdd-139ae08d75b7",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F13Small.jpg?alt=media&token=2f339eaf-7acd-4014-afe2-2c6f73e737e5",
        name: "Средство для мытья окон и стекол Clin Лимон",
        dimensionType: "мл",
        dimension: "500",
        barcode: "9000100888448",
        manufacturer: "Henkel",
        brand: "Clin",
        description:
            "Средство Clin Мультиблеск - это универсальное чистящее средство для 25 различных поверхностей. Спрей Клин дает 2 важных преимущества: блеск и защиту. Средство Клин на основе изопропилового спирта качественно очищает стекла от грязи, пыли, не оставляет разводов.",
        price: 323,
        purposes: ["Для мытья окон"]
    },
    4823002001013: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F14.jpg?alt=media&token=f1a1d8e3-9623-423e-be67-fe4692c73d51",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F14Small.jpg?alt=media&token=62d43703-52c2-4335-af53-a245d5311043",
        name: "Моющее средство Mr.Muscle для стекол и других поверхностей",
        dimensionType: "мл",
        dimension: "500",
        barcode: "4823002001013",
        manufacturer: "SC Jonhson",
        brand: "Mr.Muscle",
        description:
            "Mr. Muscle для окон Триггер После дождя 500 мл. Средство для мытья стекол и других поверхностей Mr Muscle - жидкость с ароматом для очистки оконного, витринного, автомобильного стекол, зеркал, нержавеющей стали, хромированных поверхностей.Не надо смывать. Придает блеск.",
        price: 401,
        purposes: ["Для мытья окон", "Для мытья зеркал"]
    },
    4823002000160: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F15.jpg?alt=media&token=6c09fae5-5d89-453c-ae3d-5c07688e66ff",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F15Small.jpg?alt=media&token=c3a6f3da-f2d2-422f-b481-82558cae9826",
        name: "Моющее средство Mr.Muscle для стекол Зеленый",
        dimensionType: "мл",
        dimension: "500",
        barcode: "4823002000160",
        manufacturer: "SC Jonhson",
        brand: "Mr.Muscle",
        description:
            "Благодаря входящему в состав изопропиловому спирту, чистящее средство Mr.Muscle Профессионал эффективно удаляет грязь, жир, сажу, минеральные масла, придает блеск и не оставляет разводов. Идеально подходит для мытья оконного, витринного автомобильного стекол, зеркал, кафеля, внешних панелей электробытовых приборов, хромированных поверхностей, поверхностей из нержавеющей стали, поверхности холодильника.",
        price: 318,
        purposes: ["Для мытья окон"]
    },
    4600697080710: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F16.jpg?alt=media&token=5b86dbc6-93c1-4d01-a0f5-6df64a6bb85f",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F16Small.jpg?alt=media&token=c337eb6d-ff0e-4b54-8ca5-681c72d929f7",
        name: "Чистящий порошок Сарма Антибактериальный эффект Лимон",
        dimensionType: "гр",
        dimension: "400",
        barcode: "4600697080710",
        manufacturer: "Невская Косметика",
        brand: "Сарма",
        description:
            "Прекрасно очищают посуду от налетов, жира и пригоревших остатков пищи. Обладают доказанным антибактериальным эффектом без хлора. Не содержит хлор.",
        price: 86,
        purposes: ["Для мытья посуды", "Для мытья сантехники"]
    },
    4602984004584: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F17.jpg?alt=media&token=ff7fed72-6dc4-4f90-a466-80082e4c85a7",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F17Small.jpg?alt=media&token=a0fb91cf-693d-41c6-8fa2-0f3996e41664",
        name: "Чистящее средство Sanfor Chlorum гель",
        dimensionType: "мл",
        dimension: "750",
        barcode: "4602984004584",
        manufacturer: "Ступинский хим. завод",
        brand: "Sanfor",
        description:
            "Специальная формула с содержанием хлора одинаково хорошо подходит для чистки ванн и унитазов. Эффективно удаляет серый налет, въевшуюся грязь, мыльные потеки. Мгновенно, без длительных выдерживаний, возвращает поверхностям первоначальную белизну.",
        price: 161,
        purposes: ["Для мытья сантехники"]
    },
    8000630720233: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F18.jpg?alt=media&token=b0159bc5-62e3-4fee-ad9d-9a1e20eb8d95",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F18Small.jpg?alt=media&token=db50b3fe-7207-431f-ba12-c799c395dc8d",
        name: "Чистящий спрей Cif Power & Shine Антиналет",
        dimensionType: "мл",
        dimension: "500",
        barcode: "8000630720233",
        manufacturer: "Юнилевер Русь",
        brand: "Cif",
        description:
            "Средство чистящее Легкость чистоты для ванной решит все проблемы! Чистящий спрей поможет надолго распрощаться с грязью, устаревшими пятнами от ржавчины и устойчивым налетом. Мы создаем продукты, способные облегчить вашу жизнь и сделать дом красивым и уютным без особых усилий.",
        price: 255,
        purposes: ["Для мытья сантехники", "Для мытья стен"]
    },
    4600697070681: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F19.jpg?alt=media&token=076d8d06-e832-46d0-b070-9c5d106bbcd8",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F19Small.jpg?alt=media&token=b9ccfb0d-3c9f-49e4-8eb9-5651829ca6dc",
        name: "Чистящий гель Сарма для сантехники Антиржавчина",
        dimensionType: "мл",
        dimension: "750",
        barcode: "4600697070681",
        manufacturer: "Невская Косметика",
        brand: "Сарма",
        description:
            "Предназначено для чистки сантехники, в первую очередь унитазов. Гели прекрасно справляются с ржавчиной, известковым налетом, мочевым камнем, удаляет микробы и бактерии без хлора, а также устраняют неприятные запахи.",
        price: 177,
        purposes: ["Для мытья сантехники"]
    },
    4600697080727: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F20.jpg?alt=media&token=ff2e23f2-83b2-42f5-8e23-cf789898ac23",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F20Small.jpg?alt=media&token=00565e36-7320-4132-a9fc-e4fac02d4f9a",
        name: "Чистящий порошок Сарма Антибактериальный эффект",
        dimensionType: "гр",
        dimension: "400",
        barcode: "4600697080727",
        manufacturer: "Невская Косметика",
        brand: "Сарма",
        description:
            "Средство может быть рекомендовано к использованию в детских, школьных учреждениях, санаторно-курортных учреждениях, предприятиях гостиничного хозяйства, общественного питания и иных предприятиях и организациях, где требуется очистка и антибактериальная обработка поверхностей.",
        price: 86,
        purposes: ["Для мытья посуды", "Для мытья сантехники"]
    },
    4600697080796: {
        imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F21.jpg?alt=media&token=d1c63f3d-386f-4de6-905f-f3e5f558b081",
        imgSmall:
            "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F21Small.jpg?alt=media&token=cac0413c-d333-40e8-a8a1-5fbfd431d0ce",
        name: "Чистящий гель Сарма Антибактериальный Универсал",
        dimensionType: "мл",
        dimension: "500",
        barcode: "4600697080796",
        manufacturer: "Невская Косметика",
        brand: "Сарма",
        description:
            "Предназначено для чистки керамических, эмалированных, фаянсовых поверхностей, а также поверхностей из нержавеющей стали. Отлично удаляют ржавчину, известковый налет и остатки моющих средств на кафеле и сантехнике. Придает поверхностям в ванной комнате блеск, возвращает их естественную белизну.",
        price: 144,
        purposes: ["Для мытья сантехники"]
    }
};

export const testProduct = {
    imgBig: "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F2.webp?alt=media&token=be27afb4-8f2f-4b0d-8e94-a01cf69855ed",
    imgSmall:
        "https://firebasestorage.googleapis.com/v0/b/sultan-68796.appspot.com/o/catalog%2F2Small.jpg?alt=media&token=f0138e0e-896e-4cad-bf60-d80b19a77188",
    name: "Жидкость для мытья посуды Fairy Зеленое яблоко",
    dimensionType: "мл",
    dimension: "450",
    barcode: "4015400869320",
    manufacturer: "Procter & Gamble",
    brand: "Fairy",
    description:
        "Подходит для мытья стеклянной, обычной и детской посуды, фруктов и овощей, нержавеющей стали. Fairy - это безопасный продукт, который легко смывается водой и полностью соответствует ГОСТу.",
    price: 168,
    purposes: ["Для мытья посуды", "Для мытья фруктов"]
};
