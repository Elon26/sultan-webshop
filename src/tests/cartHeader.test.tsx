import { render, screen } from "@testing-library/react";
import * as reduxHooks from "react-redux";
import CartHeader from "../components/ui/cartHeader";
import { MemoryRouter } from "react-router-dom";
import { testTotalPriceAndQuantity } from "../testsConstats";

jest.mock("react-redux");

/** Тест проверяет динамически изменяемый компонент корзины в шапке страницы на корректность отрисовки после получения данных из Redux, в т.ч. на соответствие сохранённому Snapshot'у. */
describe("CartHeader", () => {
    test("Проверка корректности блока корзины из шапки", () => {
        jest.spyOn(reduxHooks, "useSelector").mockReturnValue(
            testTotalPriceAndQuantity
        );

        const component = render(
            <MemoryRouter>
                <CartHeader windowWidth={1500} />
            </MemoryRouter>
        );

        const cartHeaderTotalPrice =
            screen.getByTestId(/cartHeaderTotalPrice/i);
        const cartHeaderTotalQuantity = screen.getByTestId(
            /cartHeaderTotalQuantity/i
        );
        expect(cartHeaderTotalPrice.textContent).toEqual("777 777 ₽");
        expect(cartHeaderTotalQuantity.textContent).toEqual("42");
        expect(component).toMatchSnapshot();
    });
});
