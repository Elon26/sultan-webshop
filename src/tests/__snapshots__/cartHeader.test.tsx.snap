// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartHeader Проверка корректности блока корзины из шапки 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="bottomHeader__item bottomHeader__cart"
        href="/sultan-webshop/cart/"
      >
        <div
          class="bottomHeader__cartImg"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 2.5A.5.5 0 0 1 .5 2H2a.5.5 0 0 1 .485.379L2.89 4H14.5a.5.5 0 0 1 .485.621l-1.5 6A.5.5 0 0 1 13 11H4a.5.5 0 0 1-.485-.379L1.61 3H.5a.5.5 0 0 1-.5-.5zM3.14 5l1.25 5h8.22l1.25-5H3.14zM5 13a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm-2 1a2 2 0 1 1 4 0 2 2 0 0 1-4 0zm9-1a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm-2 1a2 2 0 1 1 4 0 2 2 0 0 1-4 0z"
            />
          </svg>
          <span
            data-testid="cartHeaderTotalQuantity"
          >
            42
          </span>
        </div>
        <div
          class="bottomHeader__cartInfo"
        >
          <p>
            Корзина
          </p>
          <p
            data-testid="cartHeaderTotalPrice"
          >
            777 777
             ₽
          </p>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="bottomHeader__item bottomHeader__cart"
      href="/sultan-webshop/cart/"
    >
      <div
        class="bottomHeader__cartImg"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 2.5A.5.5 0 0 1 .5 2H2a.5.5 0 0 1 .485.379L2.89 4H14.5a.5.5 0 0 1 .485.621l-1.5 6A.5.5 0 0 1 13 11H4a.5.5 0 0 1-.485-.379L1.61 3H.5a.5.5 0 0 1-.5-.5zM3.14 5l1.25 5h8.22l1.25-5H3.14zM5 13a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm-2 1a2 2 0 1 1 4 0 2 2 0 0 1-4 0zm9-1a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm-2 1a2 2 0 1 1 4 0 2 2 0 0 1-4 0z"
          />
        </svg>
        <span
          data-testid="cartHeaderTotalQuantity"
        >
          42
        </span>
      </div>
      <div
        class="bottomHeader__cartInfo"
      >
        <p>
          Корзина
        </p>
        <p
          data-testid="cartHeaderTotalPrice"
        >
          777 777
           ₽
        </p>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
